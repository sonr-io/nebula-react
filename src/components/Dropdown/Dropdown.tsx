import React, { useState } from 'react';
import { DropdownProps } from '../../types/dropdownProps';
import { NebulaIcon } from '../Icons/NebulaIcon';
import { DropdownOption } from '../DropdownOption'

const cx = require('classnames');

export const Dropdown: React.FC<DropdownProps> = ({
    options
}) => {
    const [isOpen, setIsOpen] = useState(false)

    const toggleDropdown = () => setIsOpen(!isOpen)

    const dropDownClass = cx()

    return (
        <div className='relative cursor-pointer'>
            <NebulaIcon 
                onClick={toggleDropdown} 
                iconName={`${isOpen ? 'ArrowUp1' : 'ArrowDown'}`} 
                iconType='filled' 
                className={dropDownClass} 
            />
            {isOpen && (
                <div className='absolute w-52 rounded-md shadow-medium drop-shadow-medium'>
                    {(options && options.length) && (
                        options.map(({ title, leftIconName, leftIconType, rightIconName, rightIconType }, index) => {
                            const optionBorder = () => {
                                if(index === 0) return 'rounded-t-md'
                                if(index === options.length - 1) return 'rounded-b-md'
                                return ''
                            }

                            return (
                                <DropdownOption
                                    className={optionBorder()}
                                    key={index}
                                    title={title} 
                                    leftIconName={leftIconName} 
                                    leftIconType={leftIconType}
                                    rightIconName={rightIconName} 
                                    rightIconType={rightIconType} 
                                />
                            )}
                        )
                    )}
                </div>
            )}
        </div>
    )
}
