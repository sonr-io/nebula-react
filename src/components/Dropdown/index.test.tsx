import { render } from '@testing-library/react'
import { Dropdown } from "./Dropdown"

jest.mock("../../assets/outline/DocumentText1.svg", () => "document-text1-svg")
jest.mock("../../assets/filled/DocumentText1.svg", () => "document-text1-svg")
jest.mock("../../assets/duotone/DocumentText1.svg", () => "document-text1-svg")
jest.mock("../../assets/outline/Document.svg", () => "document-svg")
jest.mock("../../assets/filled/Document.svg", () => "document-svg")
jest.mock("../../assets/duotone/Document.svg", () => "document-svg")
jest.mock("../../assets/outline/ArrowDown.svg", () => "arrow-down-svg")
jest.mock("../../assets/filled/ArrowDown.svg", () => "arrow-down-svg")
jest.mock("../../assets/duotone/ArrowDown.svg", () => "arrow-down-svg")


beforeEach(() => {
    jest.resetModules();
});  

describe('Dropdown component', () => {
    test('should render', () => {
        const { asFragment }  = render(
            <Dropdown  
                options={[{
                    title: 'Option 1',
                    leftIconName: 'Document',
                    leftIconType: 'outline'
                }]}
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})