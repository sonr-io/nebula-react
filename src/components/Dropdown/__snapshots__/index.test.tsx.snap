// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render 1`] = `
<DocumentFragment>
  <div
    class="relative cursor-pointer"
  >
    <arrow-down-svg
      classname="w-4 h-4 inline w-4 h-4"
      data-testid="nebula-icon"
    />
  </div>
</DocumentFragment>
`;
